File: solution/15xxx/15927 - 회문은 회문아니야!!/15927.cpp
#include <iostream>
#include <string>
using namespace std;

#define FASTIO cin.tie(NULL); cout.tie(NULL); ios::sync_with_stdio(false);

int solve() {
    string str;
    cin >> str;
    int count = 0;
    int left = 0, right = str.size()-1;
    while(left <= right) {
        if(str[left] != str[right]) return str.size();
        if(str[0] == str[left]) count += 1;
        if(str[0] == str[right]) count += 1;
        left += 1;
        right -= 1;
    }
    if(count > str.size()) count = str.size();
    if(count == str.size()) return -1;
    else return str.size()-1;
}

int main() {
    FASTIO
    cout << solve();
}

File: solution/15xxx/15927 - 회문은 회문아니야!!/15927.py
import sys
input = sys.stdin.readline

def solve():
    word = input()[:-1]
    count = 0
    left, right = 0, len(word)-1
    while left <= right:
        if word[left] != word[right]: return len(word)
        if word[0] == word[left]: count += 1
        if word[0] == word[right]: count += 1
        left += 1
        right -= 1
    if count > len(word): count = len(word)
    if count == len(word): return -1
    else: return len(word)-1

if __name__ == '__main__':
    print(solve())